testUtils/testConnector/mod.ts

```diff
@@ -0,0 +1,20 @@
+type ConnectorSource = (nextPageToken: string) => Promise<string>
```

Comment:
It would be nice to add a test.ts where we could implement a fake source that just returns hardcoded data so we can have some unit tests for this.